scheduler/test/cook/test/mesos/quota.clj

```diff
@@ -15,64 +15,167 @@
+      (is (= 2.0 (:count (quota/get-quota (d/db conn) "fakeuser" nil)))))
+
+    (testing "retracts both count fields"
+      (quota/retract-quota! conn "default" nil "clearing default")
+      (quota/set-quota! conn "u1" nil "setting count" :count 1)
+      @(d/transact conn [[:db/add [:quota/user "u1"] :quota/count 1]])
+      (quota/retract-quota! conn "u1" nil "clearing user")
+
+      (is (= Double/MAX_VALUE (:count (quota/get-quota (d/db conn) "u1" nil)))))
+
+    (testing "set-quota! clears count field"
+      (quota/set-quota! conn "u1" nil "setup" :cpus 1.0)
+      @(d/transact conn [[:db/add [:quota/user "u1"] :quota/count 1]])
+      (is (= 1 (:quota/count (d/entity (d/db conn) [:quota/user "u1"]))))
+      (quota/set-quota! conn "u1" nil "should clear count" :cpus 2.0)
+      (is (nil? (:quota/count (d/entity (d/db conn) [:quota/user "u1"])))))))
+
+
+(deftest test-pool-support
+  (let [uri "datomic:mem://test-quota-pool-support"
+        conn (restore-fresh-database! uri)]
+    (create-pool conn "pool-1")
+    (create-pool conn "pool-2")
+    (create-pool conn "pool-3")
+    (quota/set-quota! conn "default" nil "strict" :cpus 1.0 :mem 1.0 :gpus 1.0 :count 1)
+    (quota/set-quota! conn "default" "pool-1" "lenient" :cpus 10.0 :mem 10.0 :gpus 10.0 :count 10.0)
+    (quota/set-quota! conn "u1" nil "power user" :cpus 20.0 :mem 20.0)
+    (quota/set-quota! conn "u1" "pool-1" "really lenient" :cpus 100.0 :mem 100.0)
+    (testing "get-quota, no default pool configured"
+      (let [db (mt/db conn)]
+        (is (= {:cpus 20.0 :mem 20.0 :gpus 1.0 :count 1.0} (quota/get-quota db "u1" nil)))
+        (is (= {:cpus 100.0 :mem 100.0 :gpus 10.0 :count 10.0} (quota/get-quota db "u1" "pool-1")))
+        (is (= {:cpus Double/MAX_VALUE :mem Double/MAX_VALUE :gpus Double/MAX_VALUE
+                :count Double/MAX_VALUE}
+               (quota/get-quota db "u1" "pool-2")))
+
+        (is (= {:cpus 1.0 :mem 1.0 :gpus 1.0 :count 1.0} (quota/get-quota db "u2" nil)))
+        (is (= {:cpus 10.0 :mem 10.0 :gpus 10.0 :count 10.0} (quota/get-quota db "u2" "pool-1")))))
+
+    (testing "get-quota, default pool configured"
+      (with-redefs [config/default-pool (constantly "pool-1")]
+        (let [db (mt/db conn)]
+          ; Should use the explicit defaults from pool-1
+          (is (= {:cpus 100.0 :mem 100.0 :gpus 10.0 :count 10.0} (quota/get-quota db "u1" nil)))
+          (is (= {:cpus 100.0 :mem 100.0 :gpus 10.0 :count 10.0} (quota/get-quota db "u1" "pool-1")))
+          (is (= {:cpus Double/MAX_VALUE :mem Double/MAX_VALUE :gpus Double/MAX_VALUE
+                  :count Double/MAX_VALUE}
+                 (quota/get-quota db "u1" "pool-2")))
+
+          (is (= {:cpus 10.0 :mem 10.0 :gpus 10.0 :count 10.0} (quota/get-quota db "u2" nil)))
+          (is (= {:cpus 10.0 :mem 10.0 :gpus 10.0 :count 10.0} (quota/get-quota db "u2" "pool-1"))))))
+    (testing "retract quota, no default pool configured"
```

Comment:
Nitpick: I would favor a newline before the `testing`

Reply:
Agreed, done